[Swiftify] add debug logs for safe interop

hnrklssn opened the pull request:
This adds some dynamically togglable debug logs for the `swiftify` function in the ClangImporter under `-Xllvm -debug-only=safe-interop-wrappers`, and for the `_SwiftifyImport` macro under `-Xllvm -debug-only=swiftify-import`

hnrklssn committed:
[Swiftify] add debug logs

hnrklssn commented:
@swift-ci please smoke test

j-hui reviewed (approved):
This LGTM, but I'm not sure familiar with what conventions are around inserting normally-inert debug statements like this in the Swift compiler source, so I advise waiting for at least one more approval from someone with more experience in this code base than me.

hnrklssn commented:
@swift-ci please smoke test

hnrklssn commented:
@swift-ci please smoke test

hnrklssn commented on stdlib/public/core/SwiftifyImport.swift:
```diff
@@ -62,7 +62,8 @@ public enum _SwiftifyInfo {
                              spanAvailability: String? = nil,
-                             typeMappings: [String: String] = [:]) =
+                             typeMappings: [String: String] = [:],
+                             debug: Bool = false) =
```
@DougGregor do you think including conditionally enableable debug logging in the macro is reasonable, or is that a no-no in the stdlib?

hnrklssn committed:
[Macros] Add debug logs for macro expansions

These can be enabled with `-Xcc -mllvm -debug-only=macros`.

hnrklssn committed:
[ASTPrint] Refactor CustomAttr printing (NFC)

This is a pure refactor to enable using PrintAST in the next commit.

hnrklssn committed:
[ASTPrint] Print argument list for CustomAttr

hnrklssn committed:
[ASTPrint] Only print CustomAttr arglist if PrintExprs enabled

The argument list is going to be mostly empty if PrintExprs is not
enabled. Always trying to print the arguments also broke quite a few
tests, because it would result in a mismatch between the swift interface
before and after serialization, since CustomAttr arguments are not
serialized.
